core/async/src/test_loop/event_handler.rs

```diff
@@ -89,44 +162,29 @@ impl<Event> LoopEventHandler<Vec<Event>, Event> for CaptureEvents {
 ///
 /// This is used on output events so that after the test loop finishes running
 /// we can assert on those events.
-pub fn capture_events<Event>() -> impl LoopEventHandler<Vec<Event>, Event> {
-    CaptureEvents
-}
-
-/// Implements periodic_interval().
-struct PeriodicInterval<Data, Event: Clone + PartialEq> {
-    interval: Duration,
-    event: Event,
-    delay_sender: Option<DelaySender<Event>>,
-    func: Box<dyn Fn(&mut Data) + 'static>,
-}
-
-impl<Data, Event: Clone + PartialEq> LoopEventHandler<Data, Event>
-    for PeriodicInterval<Data, Event>
-{
-    fn init(&mut self, sender: DelaySender<Event>) {
-        self.delay_sender = Some(sender);
-        self.delay_sender.as_ref().unwrap().send_with_delay(self.event.clone(), self.interval);
-    }
-
-    fn handle(&mut self, event: Event, data: &mut Data) -> Result<(), Event> {
-        if event == self.event {
-            (self.func)(data);
-            self.delay_sender.as_ref().unwrap().send_with_delay(self.event.clone(), self.interval);
-            Ok(())
-        } else {
-            Err(event)
-        }
-    }
+pub fn capture_events<Event>() -> LoopEventHandler<Vec<Event>, Event> {
+    LoopEventHandler::new_simple(|event, data: &mut Vec<Event>| data.push(event))
 }
 
 /// Periodically sends to the event loop the given event by the given interval.
 /// Each time this event is handled, the given function is called.
 /// The first invocation is triggered after the interval, not immediately.
 pub fn periodic_interval<Data, Event: Clone + PartialEq>(
```

Comment:
nit: "periodic" is kind of redundant to "interval" (at least in the context of the conventions I was referring to in the previous PR).